idl-compiler.py

```diff
@@ -480,6 +480,9 @@ class RpcVerb(ASTBase):
       to the handler function and send* method uses send_message_*_timeout
       variant of internal function to actually send the message.
       Incompatible with [[abortable]].
+    - [[cancellable]] - an additional rpc::cancellable& parameter is supplied
+      to the handler function and send* method uses send_message_*_cancellable
+      variant of internal function to actually send the message.
```

Comment:
I do not understand why do you need cancellable and abortable. You need to make existing  cancellable and with_timeout attributes composable, not introduce this new version.

Reply:
today's `[[cancellable]]` expects reference to an abort_source which is really confusing, while the new `[[cancellable]]` expects reference to a `cancellable`, and so it is composable with `[[with_timeout]]`.
Making the existing `[[cancellable]]` composable with `[[with_timeout]]` will change the derived signature in a counterintuitive way.

Reply:
I think that to keep `[[cancellable]]` we can convert the existing code to use `cancellable` object rather than an abort_source and implement a shim to subscribe to the abort_source and signal the cancellable object.
I'll see if it's feasible.

Reply:
> > I think that to keep `[[cancellable]]` we can convert the existing code to use `cancellable` object rather than an abort_source and implement a shim to subscribe to the abort_source and signal the cancellable object. I'll see if it's feasible.
> 
> Why? Forget about `cancellable` . It is RPC implementation detail that messaging service is here to hide.

For performance sensitive RPCs, if the verb is called in a coroutine, the cancellable object can be put in the coroutine frame, saving an allocation that is done today to hook the `abort_source` to the `cancellable` object.

Reply:
We can rewrite helpers with co-routines as well.